app/controllers/entries_controller.rb

```diff
@@ -31,16 +31,8 @@ def recover_entries_history(register_id, fields, params, page_size = 100)
     page = (params[:page] ||= 1).to_i
 
     if search_term.present?
-      partial = ''
-      operation_params = []
-      fields.split(',').each { |field| partial += " data->> '#{field}' ilike ? or" }
-      partial = partial[1, partial.length - 3] # Remove beginning space and end 'or'
-
-      operation_params.push(partial)
-      fields.split(',').count.times { operation_params.push("%#{search_term}%") }
-
-      query = Entry.where(register_id: register_id, entry_type: 'user').where(operation_params).order(:entry_number).reverse_order.limit(page_size).offset(page_size * (page - 1))
-      count_query = Entry.where(register_id: register_id, entry_type: 'user').where(operation_params)
+      query = Entry.where(register_id: register_id, entry_type: 'user').search_for(fields, search_term).order(:entry_number).limit(page_size).offset(page_size * (page - 1))
```

Comment:
We could benefit from some dryness here by doing

```ruby
query = Entry.where(register_id: register_id, entry_type: 'user')
```

Also, it seems that the linter is happy with these long lines but, do you find them readable? I'll admit that beyond 80 chars things start to be hard for me to follow, more than 100 it feels too much.